modules/UpgradeWizard/silentUpgrade_step1.php

```diff
@@ -541,10 +541,8 @@ function sugar_cached($file)
 
     $unzip_dir = sugar_cached('upgrades/temp');
 
-    include "$unzip_dir/manifest.php";
-    if (is_file("$unzip_dir/manifest.php")) {
-        include "{$argv[1]}/manifest.php";
-    }
+    $install_file = $sugar_config['upload_dir'] . '/upgrades/patch/' . basename($argv[1]);
+    sugar_mkdir($sugar_config['upload_dir'] . '/upgrades/patch', 0775, true);
```

Comment:
We shouldn't be setting permissions like this. I suggest using `make_writable` function which can also be improved, but still is better than using fixed permissions values directly https://github.com/salesagility/SuiteCRM/blob/master/install/install_utils.php#L1308

If an admin doesn't want to add `world` permissions, which is a good idea, and even takes the trouble of specifying this in `default_permissions` in `config.php`, who are we to then go open up his system like this?

Reply:
@pgorod Totally agreed, this is a hacky fix ontop of a hacky fix intended to just "force" the upgrade to work. We actually do an absolute ton of other things forcefully and without warning regarding permissions and files during the upgrade process (E.g. just uploading a pack, even the wrong one, will steamroll force replace all the DBManagers and a couple dozen other files).

I'm rather reluctant to change anything here since it requires so much testing upgrading legacy instances with customisations.

Reply:
Anyway, it seems `sugar_mkdir` will happily ignore whatever you pass it there and just use the value from `config.php`.
See #2810

Go ahead with whatever you think is best at this point. I agree that a proper fix needs to go deeper than this, and involves a lot more changes. Sigh...

Reply:
> Go ahead with whatever you think is best at this point. I agree that a proper fix needs to go deeper than this, and involves a lot more changes. Sigh...

Just to note, I'll be making another PR (Will likely need to go to develop though) after the next release that will fix all of these problems and bring the upgrade time down to 30 or so seconds :).

Reply:
In that case, to add to your work, I have a couple more requests

- a check that warns users that they are trying to use a full installation pack, or an add-on, in the upgrade wizard (instead of just saying it's invalid).

- a button to upgrade directly from the Internet, avoiding the download-then-upload loop. Just "Upgrade available, click here to apply it".

Of course you won't have time for all this but I thought I might as well ask for it!

Reply:
Done and done :D. The upgrade from the internet I was going to split into a third PR though.

Reply:
That's I call _agile_ 😱